source/dub/dub.d

```diff
@@ -813,7 +830,7 @@ class Dub {
 			import std.zip : ZipException;
 
 			auto path = getTempFile(basePackageName, ".zip");
-			supplier.fetchPackage(path, basePackageName, dep, (options & FetchOptions.usePrerelease) != 0); // Q: continue on fail?
+			supplier.fetchPackage(path, basePackageName, Dependency(range), (options & FetchOptions.usePrerelease) != 0); // Q: continue on fail?
```

Comment:
shouldn't the suppliers also be updated?

Reply:
I can't because it's an `interface`, so it would potentially break library users.
The `PackageSupplier` has many issues (tied to Json, this, and generally too tailored to the registry) and I plan to overhaul it in the future, but I'd rather do it in one pass.